purchases/src/main/kotlin/com/revenuecat/purchases/utils.kt

```diff
@@ -193,6 +193,9 @@ internal fun String.sha1() =
         }
 
 internal fun JSONObject.getNullableString(name: String): String? = this.getString(name).takeUnless { this.isNull(name) }
+internal fun JSONObject.optNullableString(name: String): String? = this.optString(name)?.takeUnless {
```

Comment:
I think `optString` returns `""` if the JSON doesn't contain the name, which is very very annoying

Reply:
good catch. I got bit by looking at the search result
<img width="640" alt="image" src="https://user-images.githubusercontent.com/3922667/85208695-d69dd680-b308-11ea-95a7-eac53b56cf2c.png">

but not reading in the actual doc
> Returns the value mapped by name if it exists, coercing it if necessary, or the empty string if no such mapping exists.
Fixing

Reply:
the method actually does work, though, because of the `takeUnless`, which checks if the object `isNull`. That check will return true if the key isn't found in the JSON https://developer.android.com/reference/org/json/JSONObject#isNull(java.lang.String)

Reply:
I added an extra test case to make sure (also tested manually)

Reply:
I see. `optString` will never return `null` though, so `?.` should be `.` right?